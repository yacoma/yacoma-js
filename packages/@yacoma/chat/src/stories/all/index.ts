export * from './Explorer'
// export * from './Header'
// export * from './Chat'
// export * from './People'
// export * from './ContactsWidget'
//
