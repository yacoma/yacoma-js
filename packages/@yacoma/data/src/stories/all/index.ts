export * from './ContactPrint'
export * from './Loading'
export * from './RegisterName'
export * from './RegisterVerify'
export * from './RegisterWords'
export * from './WordLogin'
