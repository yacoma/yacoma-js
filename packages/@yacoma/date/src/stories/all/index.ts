export * from './Date'
export * from './RelativeDate'
export * from './DatePicker'
