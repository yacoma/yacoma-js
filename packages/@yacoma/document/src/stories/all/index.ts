export * from './ExplorerWidget'
export * from './Explorer'
export * from './UserAccessView'
export * from './ItemTitle'
export * from './ItemIcon'
export * from './ItemDrag'
export * from './DocumentLink'
export * from './DocumentEditor'
