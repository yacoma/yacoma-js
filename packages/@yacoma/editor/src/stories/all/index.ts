export * from './Toolbox'
export * from './Editor'
export * from './CustomParagraph'
