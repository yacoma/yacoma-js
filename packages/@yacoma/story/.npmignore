/node_modules
/coverage
/setup
/src
/.cache
.gitignore
.npmignore
.prettierrc.json
.travis.yml
tsconfig.json
tslint.json
package-lock.json
yarn.lock