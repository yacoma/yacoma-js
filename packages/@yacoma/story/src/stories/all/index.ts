export * from './actions'
export * from './children'
export * from './props'
export * from './state'
export * from './storiesDefaults'
