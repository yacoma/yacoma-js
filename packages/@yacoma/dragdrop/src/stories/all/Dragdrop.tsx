import * as React from 'react'
import { Stories } from '@yacoma/story'
import { Drag, DraggingProps as Props } from '../..'
import { useOvermind } from '../../app'
import { dropStyles } from '../../components'
import { fileDrop } from '../../fileDrop'
import { Comp, config, Doc, Group, Group2, styled } from '../helpers'

const component = styled(Drag)`
  background: rgba(255, 255, 255, 0.8);
`
const FileDropDiv = styled.div`
  margin: 10px;
  padding: 10px;
  width: 100px;
  height: 50px;
  ${dropStyles};
`

const filedrop: Comp = () => {
  const ctx = useOvermind()
  const ref = React.useRef(null)

  return (
    <FileDropDiv
      {...fileDrop(ctx, {
        ref,
        onDrop(args) {
          console.log('dropped', args)
        },
        accept(file) {
          console.log('accept?', file)
          return true
        },
        payload: { collectionId: 'someCollectionId' },
      })}
    >
      FILE DROP HERE
    </FileDropDiv>
  )
}

export const dragdropStories: Stories<Props> = {
  config,
  component,
  name: 'Dragdrop',
  stories: [
    {
      name: 'dragging',
      props: {},
      state: {
        dragdrop: {
          drag: {
            type: 'doc',
            anchor: { x: 10, y: 10 },
            payload: { name: 'Lily', age: 19 },
          },
          position: { x: 320, y: 90 },
        },
      },
    },

    {
      name: 'start drag',
      props: {},
      wrapper: ({ children }: any) => (
        <div>
          {children}
          <Doc name="My name" group="dummy" />
        </div>
      ),
    },

    {
      name: 'file drop',
      component: filedrop,
    },

    {
      name: 'drag & drop',
      props: {},
      wrapper: ({ children }: any) => (
        <div>
          {children}
          {/* Details of how the drop works are in ../helpers/Group */}
          <Group group="foo" />
          <Group group="bar" />
          <Group2 group="baz" />
        </div>
      ),
      state: {
        groups: {
          foo: {
            hop: 'hop',
            lala: 'lala',
          },
          bar: {
            bar: 'bar',
          },
          baz: {},
        },
      },
    },
  ],
}
