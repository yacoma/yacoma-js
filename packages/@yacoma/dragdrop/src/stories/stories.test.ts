import { testStories } from '@yacoma/story'
import * as stories from './all'

testStories({}, stories)
