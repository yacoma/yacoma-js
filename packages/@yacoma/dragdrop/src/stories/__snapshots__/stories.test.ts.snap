// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dialog (click to view) should render drag & drop 1`] = `
<div
  className="theme-3"
>
  <div
    className="sc-EHOje jwoVsz"
  >
    <div
      className="sc-bxivhb ckTHop"
    >
      <div
        className="sc-ifAKCX dakLHt"
      >
        <div
          className="Title sc-bwzfXH bzeJnn"
        >
          drag & drop
        </div>
        <div
          className="sc-bdVaJa gdMgPe"
        >
          &lt;Drag /&gt;
          <div
            className="jsx"
          >
            &lt;Drag /&gt;
          </div>
        </div>
      </div>
      <div>
        <div
          className="sc-jTzLTM fDjupc"
        >
          <h3>
            (Group: 
            foo
            )
          </h3>
          <div
            className="sc-gqjmRU ejrjCJ"
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseMove={[Function]}
            onMouseUp={[Function]}
          >
            
            hop
          </div>
          <div
            className="sc-gqjmRU ejrjCJ"
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseMove={[Function]}
            onMouseUp={[Function]}
          >
            
            lala
          </div>
        </div>
        <div
          className="sc-jTzLTM fDjupc"
        >
          <h3>
            (Group: 
            bar
            )
          </h3>
          <div
            className="sc-gqjmRU ejrjCJ"
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseMove={[Function]}
            onMouseUp={[Function]}
          >
            
            bar
          </div>
        </div>
        <div
          className="sc-fjdhpX lguhMC"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <h3>
            (Group2: 
            baz
            )
          </h3>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Dialog (click to view) should render dragging 1`] = `
<div
  className="theme-1"
>
  <div
    className="sc-EHOje jwoVsz"
  >
    <div
      className="sc-bxivhb ckTHop"
    >
      <div
        className="sc-ifAKCX dakLHt"
      >
        <div
          className="Title sc-bwzfXH bzeJnn"
        >
          dragging
        </div>
        <div
          className="sc-bdVaJa gdMgPe"
        >
          &lt;Drag /&gt;
          <div
            className="jsx"
          >
            &lt;Drag /&gt;
          </div>
        </div>
      </div>
      <div
        className="sc-htpNat eIswBe"
      >
        <div
          className="sc-jzJRlG fSwYev"
          style={
            Object {
              "left": 310,
              "top": 80,
            }
          }
        >
          <div
            className="sc-gqjmRU ejrjCJ"
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseMove={[Function]}
            onMouseUp={[Function]}
          >
            dragging 
            Lily
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Dialog (click to view) should render start drag 1`] = `
<div
  className="theme-2"
>
  <div
    className="sc-EHOje jwoVsz"
  >
    <div
      className="sc-bxivhb ckTHop"
    >
      <div
        className="sc-ifAKCX dakLHt"
      >
        <div
          className="Title sc-bwzfXH bzeJnn"
        >
          start drag
        </div>
        <div
          className="sc-bdVaJa gdMgPe"
        >
          &lt;Drag /&gt;
          <div
            className="jsx"
          >
            &lt;Drag /&gt;
          </div>
        </div>
      </div>
      <div>
        <div
          className="sc-gqjmRU ejrjCJ"
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseMove={[Function]}
          onMouseUp={[Function]}
        >
          
          My name
        </div>
      </div>
    </div>
  </div>
</div>
`;
